### PR TITLE
[BUGFIX] Using a RuntimeBatchRequest in a Checkpoint at runtime with Checkpoint.run()

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- GREAT-384
- Subtasks: GREAT-368, GREAT-369

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/checkpoint/test_checkpoint.py [hunk 1]
```diff
@@ -1213,10 +1213,6 @@ def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_
 
     assert len(context.validations_store.list_keys()) == 1
     assert results["success"] == True
-    try:
```
Comment (Shinnnyshinshin) [inline tests/checkpoint/test_checkpoint.py:1216]: Thank you for cleaning this pattern up @NathanFarmer.. btw do know what this try-except was trying to catch in our tests?

File: tests/conftest.py [hunk 2]
```diff
@@ -2322,6 +2322,7 @@ def titanic_pandas_data_context_with_v013_datasource_with_checkpoints_v1_with_em
 def titanic_spark_data_context_with_v013_datasource_with_checkpoints_v1_with_empty_store_stats_enabled(
     tmp_path_factory,
     monkeypatch,
+    spark_session,
```
Comment (Shinnnyshinshin) [inline tests/conftest.py:2325]: :)

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you so much for such a thorough PR.  One non-blocking question for me regarding the try-except pattern that was eliminated

File: tests/checkpoint/test_checkpoint.py (hunk 1 above)
Comment (NathanFarmer) [reply tests/checkpoint/test_checkpoint.py:1216]: There was no custom `__repr__` prior to PR #3680 and it was failing due to it not knowing how to represent DataFrames as strings.

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**